dd-sdk-android/src/test/kotlin/com/datadog/android/rum/internal/domain/scope/RumViewManagerScopeTest.kt

```diff
@@ -663,52 +636,40 @@ internal class RumViewManagerScopeTest {
     // region ApplicationStarted
 
     @Test
-    fun `𝕄 send ApplicationStarted event 𝕎 onViewDisplayed() {Legacy}`(
-        forge: Forge,
-        @IntForgery(min = Build.VERSION_CODES.KITKAT, max = Build.VERSION_CODES.N) apiVersion: Int
+    fun `𝕄 send ApplicationStarted event 𝕎 handleEvent ()`(
+        forge: Forge
     ) {
         // Given
-        CoreFeature.processImportance = RunningAppProcessInfo.IMPORTANCE_FOREGROUND
-        whenever(mockBuildSdkVersionProvider.version()) doReturn apiVersion
-        val childView: RumViewScope = mock()
-        val startViewEvent = forge.startViewEvent()
-
-        // When
-        testedScope.onViewDisplayed(startViewEvent, childView, mockWriter)
-
-        // Then
-        argumentCaptor<RumRawEvent> {
-            verify(childView).handleEvent(capture(), same(mockWriter))
-
-            val event = firstValue as RumRawEvent.ApplicationStarted
-            assertThat(event.applicationStartupNanos).isEqualTo(RumFeature.startupTimeNs)
+        // Because we have to test that the `application_started` action is written, we need to set
+        // up the feature scope properly
+        val mockRumFeatureScope: FeatureScope = mock()
+        val mockEventBatchWriter: EventBatchWriter = mock()
+        whenever(mockSdkCore.getFeature(RumFeature.RUM_FEATURE_NAME)) doReturn mockRumFeatureScope
+        whenever(mockRumFeatureScope.withWriteContext(any(), any())) doAnswer {
+            val callback = it.getArgument<(DatadogContext, EventBatchWriter) -> Unit>(1)
+            callback.invoke(fakeDatadogContext, mockEventBatchWriter)
         }
-        verifyZeroInteractions(mockWriter)
-    }
-
-    @Test
-    fun `𝕄 send ApplicationStarted event 𝕎 onViewDisplayed() {Nougat+}`(
-        forge: Forge,
-        @IntForgery(min = Build.VERSION_CODES.N) apiVersion: Int
-    ) {
-        // Given
+        val appStartTimeNs = System.nanoTime() - TimeUnit.MILLISECONDS.toNanos(500)
+        whenever(mockAppStartTimeProvider.appStartTimeNs) doReturn appStartTimeNs
         CoreFeature.processImportance = RunningAppProcessInfo.IMPORTANCE_FOREGROUND
-        whenever(mockBuildSdkVersionProvider.version()) doReturn apiVersion
-        val childView: RumViewScope = mock()
-        val startViewEvent = forge.startViewEvent()
+
+        val fakeEvent = forge.addErrorEvent()
```

Comment:
we need to make sure that it is working with any event, so I would suggest to add a method in the `RumRawEventExt` file which will return any `RumRawEvent` available and use it here.